### PR TITLE
mercury: add v2.3.0

### PR DESCRIPTION
**Pull request by soumagne**
* fix Boost dependency

* fix testing and move psm2/sockets as unsupported

### CODE REVIEW

File: var/spack/repos/builtin/packages/mercury/package.py [hunk 1]
```diff
@@ -16,6 +15,7 @@ class Mercury(CMakePackage):
     maintainers("soumagne")
     tags = ["e4s"]
     version("master", branch="master", submodules=True)
+    version("2.3.0", sha256="e9e62ce1bb2fd482f0e85ad75fa255d9750c6fed50ba441a03de93b3b8eae742")
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/mercury/package.py]:
```suggestion
    maintainers("soumagne")
    
    tags = ["e4s"]
    
    version("master", branch="master", submodules=True)
    version("2.3.0", sha256="e9e62ce1bb2fd482f0e85ad75fa255d9750c6fed50ba441a03de93b3b8eae742")
```

@soumagne thanks for the PR! This isn't something you touched but just to bring the package up in line with the common styling of others, can we add a newline between the maintainers, tags, and versions?

Comment (soumagne) [reply var/spack/repos/builtin/packages/mercury/package.py]: ok done

Comment (alecbcs) [reply var/spack/repos/builtin/packages/mercury/package.py]: Great! Thank you!

**Review by alecbcs — APPROVED** [review]
Looks good to me!

**Pull request merged**